pkg/operator/ceph/object/spec.go

```diff
@@ -540,8 +558,34 @@ func (c *clusterConfig) generateVolumeSourceWithTLSSecret() (*v1.SecretVolumeSou
 	return secretVolSrc, nil
 }
 
-func (c *clusterConfig) rgwTLSSecretType() (v1.SecretType, error) {
-	rgwTlsSecret, err := c.context.Clientset.CoreV1().Secrets(c.clusterInfo.Namespace).Get(context.TODO(), c.store.Spec.Gateway.SSLCertificateRef, metav1.GetOptions{})
+func (c *clusterConfig) generateVolumeSourceWithCaBundleSecret() (*v1.SecretVolumeSource, error) {
+	// Keep the ca-bundle as secure as possible in the container. Give only user read perms.
+	// Same as above for generateVolumeSourceWithTLSSecret function.
+	userReadOnly := int32(0400)
+	var caBundleVolSrc *v1.SecretVolumeSource
+	// TODO: add support for OpenShift based on ConfigMapVolumeSource and key `service.beta.openshift.io/inject-cabundle`
```

Comment:
The cabundle can be injected from a configmap? Does it not contain a private key?

Reply:
Well, this is what i got in OpenShift documentation - https://docs.openshift.com/container-platform/4.6/security/certificates/service-serving-certificate.html#add-service-certificate-configmap_service-serving-certificate - link is placed here https://rook.io/docs/rook/v1.6/ceph-object-store-crd.html#gateway-settings , for service parameter

Reply:
Got it, thanks for the link

Reply:
Before we merge, how about if the TODO is removed and an issue is opened to track that work item if needed?